src/test/regress/sql/sequential_ddl.sql

```diff
@@ -0,0 +1,114 @@
+--
+CREATE SCHEMA test_seq_ddl;
+SET search_path TO 'test_seq_ddl';
+
+-- disbable 2PC recovery since our tests will check that
+ALTER SYSTEM SET citus.recover_2pc_interval TO -1;
+SELECT pg_reload_conf();
+
+CREATE TABLE test_table(a int, b int);
+SELECT create_distributed_table('test_table', 'a');
+
+-- we should see #worker transactions
+-- when sequential mode is used
+SET citus.multi_shard_modify_mode TO 'sequential';
+SELECT recover_prepared_transactions();
+ALTER TABLE test_table ADD CONSTRAINT a_check CHECK(a > 0);
+SELECT tx_count = worker_count FROM (SELECT count(*) as  tx_count FROM pg_dist_transaction WHERE gid LIKE 'citus_%_' || pg_backend_pid() || '%_%') as s1,  (SELECT count(*) as worker_count FROM pg_dist_node WHERE noderole = 'primary') as s2;
+
+-- we should see placement count # transactions
+-- when parallel mode is used
+SET citus.multi_shard_modify_mode TO 'parallel';
+SELECT recover_prepared_transactions();
+ALTER TABLE test_table ADD CONSTRAINT b_check CHECK(b > 0);
+SELECT count(*) = current_setting('citus.shard_count')::bigint * current_setting('citus.shard_replication_factor')::bigint FROM pg_dist_transaction WHERE gid LIKE 'citus_%_' || pg_backend_pid() || '%_%'; 
+
+-- with 1PC, we should not see and distributed TXs in the pg_dist_transaction
+SET citus.multi_shard_commit_protocol TO '1pc';
+SET citus.multi_shard_modify_mode TO 'sequential';
+SELECT recover_prepared_transactions();
+ALTER TABLE test_table ADD CONSTRAINT c_check CHECK(a > 0);
+SELECT tx_count = 0 FROM (SELECT count(*) as  tx_count FROM pg_dist_transaction WHERE gid LIKE 'citus_%_' || pg_backend_pid() || '%_%') as s1;
+
+SET citus.multi_shard_commit_protocol TO '1pc';
+SET citus.multi_shard_modify_mode TO 'parallel';
+SELECT recover_prepared_transactions();
+ALTER TABLE test_table ADD CONSTRAINT d_check CHECK(a > 0);
+SELECT tx_count = 0 FROM (SELECT count(*) as  tx_count FROM pg_dist_transaction WHERE gid LIKE 'citus_%_' || pg_backend_pid() || '%_%') as s1;
+
+CREATE TABLE ref_test(a int);
+SELECT create_reference_table('ref_test');
+SET citus.multi_shard_commit_protocol TO '1pc';
+
+-- reference tables should always use 2PC
+SET citus.multi_shard_modify_mode TO 'sequential';
+SELECT recover_prepared_transactions();
+CREATE INDEX ref_test_seq_index ON ref_test(a);
+SELECT tx_count = worker_count FROM (SELECT count(*) as  tx_count FROM pg_dist_transaction WHERE gid LIKE 'citus_%_' || pg_backend_pid() || '%_%') as s1,  (SELECT count(*) as worker_count FROM pg_dist_node WHERE noderole = 'primary') as s2;
+
+-- reference tables should always use 2PC
+SET citus.multi_shard_modify_mode TO 'parallel';
+SELECT recover_prepared_transactions();
+CREATE INDEX ref_test_seq_index_2 ON ref_test(a);
+SELECT tx_count = worker_count FROM (SELECT count(*) as  tx_count FROM pg_dist_transaction WHERE gid LIKE 'citus_%_' || pg_backend_pid() || '%_%') as s1,  (SELECT count(*) as worker_count FROM pg_dist_node WHERE noderole = 'primary') as s2;
```

Comment:
Is there a way to make these more readable?

Reply:
Initially I started with a single example, now ended up with many. So, lots of copy&paste.

Maybe wrapping the relevant queries into a functions makes it better?
```SQL
CREATE OR REPLACE FUNCTION distributed_2PCs_are_equal_to_worker_count()
    RETURNS bool AS
$$
DECLARE
    result bool;
BEGIN
    SELECT tx_count = worker_count FROM (SELECT count(*) as  tx_count FROM pg_dist_transaction WHERE gid LIKE 'citus_%_' || pg_backend_pid() || '%_%') as s1,  (SELECT count(*) as worker_count FROM pg_dist_node WHERE noderole = 'primary') as s2 INTO result;
    RETURN result;
END;
$$
LANGUAGE 'plpgsql' IMMUTABLE;

CREATE OR REPLACE FUNCTION distributed_2PCs_are_equal_to_placement_count()
    RETURNS bool AS
$$
DECLARE
    result bool;
BEGIN
    SELECT count(*) = current_setting('citus.shard_count')::bigint * current_setting('citus.shard_replication_factor')::bigint FROM pg_dist_transaction WHERE gid LIKE 'citus_%_' || pg_backend_pid() || '%_%'
     INTO result;
    RETURN result;
END;
$$
LANGUAGE 'plpgsql' IMMUTABLE;

```